### PR TITLE
Fix stack overflow in SSHCredentials constructor

### LINKED ISSUE #21775: LibGit2 Stackoverflow
```julia
julia> LibGit2.SSHCredentials("blah", "blah")
ERROR: StackOverflowError:

julia> versioninfo()
Julia Version 0.6.0-rc1.0
Commit 6bdb395* (2017-05-07 00:00 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Prescott)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, broadwell)
```

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #21775

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -1689,3 +1689,15 @@ mktempdir() do dir
         end
     end
 end
+
+@testset "Issue #21775" begin
+    okay = true
+    try
+        cred = LibGit2.SSHCredentials("rick", "astley")
```
Comment (kshyatt) [inline test/libgit2.jl]: nice

**Review by kshyatt — APPROVED** [review]
LGTM but you might want @Keno or @omus to take a look as well.

File: test/libgit2.jl [hunk 2]
```diff
@@ -1689,3 +1689,15 @@ mktempdir() do dir
...
+    catch err
+        okay = false
+        err isa StackOverflowError || rethrow(err)
+    finally
+        @test okay
+    end
+end
```
Comment (omus) [inline test/libgit2.jl]: I'm not sure we need to explicitly be testing for a StackOverflow. Maybe just a basic test which uses this constructor should be added into the "Credentials" testset?

**Review by omus — CHANGES_REQUESTED** [review]
The fix looks good. I accidentally introduced this in #20738.

File: test/libgit2.jl (hunk 2 above)
Comment (ararslan) [inline test/libgit2.jl]: Good idea. I was just being lazy here. Implemented your suggestion. Thanks!

**Comment by ararslan** [comment]
64-bit AV failure is unrelated

**Review by omus — APPROVED** [review]

**Pull request merged**